e2e/events/test_events_app.py

```diff
@@ -0,0 +1,45 @@
+from conftest import ShinyAppProc, create_example_fixture
+from playwright.sync_api import Page, ConsoleMessage
+from controls import ActionButton, UIOutput, NavControls
+
+event_app = create_example_fixture("event")
+
+
+def test_event(page: Page, event_app: ShinyAppProc):
+    page.goto(event_app.url)
+
+    # sync
+    click_me_sync = ActionButton(page, "btn")
+    click_me_sync.loc.click()
+
+    output1 = UIOutput(page, "btn_value")
+    assert output1.get_text() == "1"
```

Comment:
@schloerke and I realized this week that `assert output1.get_text() == "1"` is not the same as `expect(output1.loc).to_have_text("1")`. In the latter case, Playwright will keep retrying the assertion for up to 5 seconds (by default) until it passes; in the former case, it will only try one. See https://playwright.dev/python/docs/test-assertions

I think this greatly affects the way we need to write these controls, any approach where we retrieve a value out of the control and do assertions on it will likely be flawed.

Reply:
Definitely. I learned more about this and agree `expect` is the more stable way to do these checks. I will update any tests that use `assert`. Thanks again for all the feedback.